sdk/storage/Azure.Storage.Files.Shares/src/ShareClient.cs

```diff
@@ -220,11 +230,48 @@ public ShareClient(Uri shareUri, StorageSharedKeyCredential credential, ShareCli
         /// <remarks>
         /// This constructor should only be used when shared access signature needs to be updated during lifespan of this client.
         /// </remarks>
-        public ShareClient(Uri shareUri, AzureSasCredential credential, ShareClientOptions options = default)
+        public ShareClient(
+            Uri shareUri,
+            AzureSasCredential credential,
+            ShareClientOptions options = default)
             : this(shareUri, credential.AsPolicy<ShareUriBuilder>(shareUri), options, sasCredential: credential)
         {
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ShareClient"/>
+        /// class.
+        /// </summary>
+        /// <param name="shareUri">
+        /// A <see cref="Uri"/> referencing the share that includes the
+        /// name of the account and the name of the share.
+        /// </param>
+        /// <param name="credential">
+        /// The token credential used to sign requests.
+        /// </param>
+        /// <param name="fileRequestIntent">
+        /// File request intent.
+        /// </param>
+        /// <param name="options">
+        /// Optional client options that define the transport pipeline
+        /// policies for authentication, retries, etc., that are applied to
+        /// every request.
+        /// </param>
+        public ShareClient(
+            Uri shareUri,
+            TokenCredential credential,
+            ShareFileRequestIntent? fileRequestIntent = default,
```

Comment:
TODO consider defaulting this to `FileRequestIntent.Backup` so it can be abstracted out from the user.

Reply:
Leave this here.